R/bm-tpc-h.R

```diff
@@ -639,6 +639,83 @@ tpc_h_queries[[10]] <- function(input_func) {
     collect()
 }
 
+tpc_h_queries[[11]] <- function(input_func) {
+  # SELECT
+  #     ps_partkey,
+  #     sum(ps_supplycost * ps_availqty) AS value
+  # FROM
+  #     partsupp,
+  #     supplier,
+  #     nation
+  # WHERE
+  #     ps_suppkey = s_suppkey
+  #     AND s_nationkey = n_nationkey
+  #     AND n_name = 'GERMANY'
+
+  partsupp <- input_func("partsupp")
+  supplier <- input_func("supplier")
+  nation <- input_func("nation")
+
+  joined_filtered <- partsupp %>%
+    inner_join(supplier, by = c("ps_suppkey" = "s_suppkey")) %>%
+    inner_join(nation, by = c("s_nationkey" = "n_nationkey")) %>%
+    filter(n_name == "GERMANY")
+
+  # GROUP BY
+  #     ps_partkey
+  # HAVING
+  #     sum(ps_supplycost * ps_availqty) > (
+  #         SELECT
+  #             sum(ps_supplycost * ps_availqty) * 0.0001000000
+  #         FROM
+  #             partsupp,
+  #             supplier,
+  #             nation
+  #         WHERE
+  #             ps_suppkey = s_suppkey
+  #             AND s_nationkey = n_nationkey
+  #             AND n_name = 'GERMANY')
+  # ORDER BY
+  #     value DESC;
+
+  # grouped mutate/filter works in dplyr but errors in Arrow because
+  # window functions aren't currently supported
+  # result1 <- joined_filtered %>%
+  #   mutate(global_value = sum(ps_supplycost * ps_availqty) * 0.0001000000) %>%
+  #   group_by(ps_partkey) %>%
+  #   mutate(value = sum(ps_supplycost * ps_availqty)) %>%
+  #   filter(value > global_value) %>%
+  #   ungroup() %>%
+  #   arrange(desc(value)) %>%
+  #   select(-global_value) %>%
+  #   collect()
+
+  # ...but not in Arrow yet, so we do a summarize() + left_join()
+  global_agr <- joined_filtered %>%
+    mutate(global_agr_key = 1L) %>%
+    group_by(global_agr_key) %>%
+    summarise(
+      global_value = sum(ps_supplycost * ps_availqty) * 0.0001000000
```

Comment:
It's possible we're running into floating point precision issues here. The TPC-H spec is all-decimal, though our support for that from R is not super great. You might try `Scalar$create(0.0001000000)$cast(decimal(10, 10))`

```
format(0.0001000000, digits = 22)
[1] "0.0001000000000000000047922"
```

When I ran into this before, I ended up [using round](https://github.com/ursacomputing/arrowbench/blob/main/R/bm-tpc-h.R#L404-L407) though looking back, probably should have been more direct like this!